See also [Arch] and [Gentoo] for similar patches in other distros.
[Arch] https://github.com/archlinux/svntogit-community/blob/packages/renpy/trunk/renpy-system-fribidi.patch
[Gentoo] https://gitweb.gentoo.org/repo/gentoo.git/tree/games-engines/renpy/files/renpy-7.3.5-use-system-fribidi.patch

Index: renpy-7.4.0-source/module/renpybidicore.c
===================================================================
--- renpy-7.4.0-source.orig/module/renpybidicore.c
+++ renpy-7.4.0-source/module/renpybidicore.c
@@ -1,9 +1,3 @@
 #include <Python.h>
-
-#ifdef RENPY_BUILD
 #include <fribidi.h>
-#else
-#include <fribidi-src/lib/fribidi.h>
-#endif
-
 #include <stdlib.h>

 #ifndef alloca
Index: renpy-7.4.0-source/module/setup.py
===================================================================
--- renpy-7.4.0-source.orig/module/setup.py
+++ renpy-7.4.0-source/module/setup.py
@@ -125,30 +125,13 @@ cython(
     sdl + [ png, 'z', 'm' ])

 FRIBIDI_SOURCES = """
-fribidi-src/lib/fribidi.c
-fribidi-src/lib/fribidi-arabic.c
-fribidi-src/lib/fribidi-bidi.c
-fribidi-src/lib/fribidi-bidi-types.c
-fribidi-src/lib/fribidi-deprecated.c
-fribidi-src/lib/fribidi-joining.c
-fribidi-src/lib/fribidi-joining-types.c
-fribidi-src/lib/fribidi-mem.c
-fribidi-src/lib/fribidi-mirroring.c
-fribidi-src/lib/fribidi-run.c
-fribidi-src/lib/fribidi-shape.c
 renpybidicore.c
 """.split()
 cython(
     "_renpybidi",
     FRIBIDI_SOURCES,
-    includes=[
-        BASE + "/fribidi-src/",
-        BASE + "/fribidi-src/lib/",
-        ],
-    define_macros=[
-        ("FRIBIDI_ENTRY", ""),
-        ("HAVE_CONFIG_H", "1"),
-        ])
+    includes=["/usr/include/fribidi"],
+    libs=["fribidi"])


 cython("_renpysteam", language="c++", compile_if=steam_sdk, libs=["steam_api"])
